### ISSUE TITLE
BUG: Extremely slow performance (affecting Obsidian itself) after opening existing (2.0.25) drawings with 2.1.4

### ISSUE THREAD

**Issue by wesleyverbruggen** [zsviczian/obsidian-excalidraw-plugin#1738]
Thank you for developing and maintaining this great plugin. I'm encountering a critical bug forcing me to quit Obsidian to be able to reach other pages, since Obsidian is lagging so hard after opening older Excalidraw drawings. 

--------

**Your environment**
SYSTEM INFO:
	Obsidian version: v1.5.12
	Installer version: v1.4.13
	Operating system: Windows 10 Home 10.0.22631
	Login status: logged in
	Catalyst license: none
	Insider build toggle: off
	Live preview: off
	Base theme: adapt to system
	Community theme: Minimal v7.4.8
	Snippets enabled: 12
	Restricted mode: off
	Plugins installed: 17
	Plugins enabled: 12
		1: Dataview v0.5.66
		2: Desmos v0.6.6
		3: Excalidraw v2.1.4
		4: Homepage v3.8.0
		5: Metadata Menu v0.8.7
		6: Style Settings v1.0.8
		7: Tasks v7.0.0
		8: Latex Suite v1.9.2
		9: Contextual Typography v2.2.5
		10: QuickAdd v1.8.1
		11: Commander v0.5.1
		12: Minimal Theme Settings v7.5.0

**Describe the bug**
Everything works fine starting Obsidian. Displaying embedded drawings is no problem. Creating new drawings also works great. Opening old drawings is very quick, but the the Excalidraw window freezes immediately after. Closing the drawing returns Obsidian to its light and usable state.

**To Reproduce**
1. Go to a note with a previously added drawing / existing drawing. Everything works as before.
2. Open the drawing by double clicking it.
3. Try to use it. Not possible without extreme (30-50 seconds) of lag.
4. Close the drawing to make Obsidian work normal again.

**Expected behavior**
No lag after opening the Excalidraw drawing.

Please let me know if I can provide more information.

**Comment by wesleyverbruggen** [comment]
The issue remains if I copy the content of the old drawing to a newly created drawing, so the problem is larger than I thought.

**Comment by zsviczian** [comment]
I don't experience the same.
Can I ask you to disable all other plugins and switch to Obsidian default theme, restart Obsidian and test again?

**Comment by zsviczian** [comment]
Does this happen to all your drawings, or just some in particular?
can you share a sample that is slow?

**Comment by wesleyverbruggen** [comment]
Hi zsviczian, thank you for your swift reply. I tried it and selectively added plugins until the Excalidraw window started lagging again. It seems like the update to Metadata Menu caused it. On the GitHub page of Metadata Menu I found a similar issue regarding .pdf files.

> **UPDATE** After looking around the settings I found the culprit of the lag In the settings > Meta Menu button > Properties
> 
> Unchecking this checkbox removed the lag entirely when opening PDFs. As of why it does that, I could not say.

That fixes it for me too. My sincere apologies for adding the issue here. I will be more specific in finding the root cause of a problem the next time. Thank you for the help!
